### PR TITLE
Normal obj

### PR DESCRIPTION
**Pull request by JacquesOlivierLachaud**
# PR Description

Add vector field export to OBJ in Shortcuts module.

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+
+## Bug Fixes
+
+- *Mathematics*
+  - Put SimpleMatrix * scalar operation in DGtal namespace (Jacques-Olivier Lachaud,
+    [#1412](https://github.com/DGtal-team/DGtal/pull/1412))
+
+- *Helpers*
+  - Fix Metric problem due to implicit RealPoint toward Point conversion when computing
+    convolved trivial normals in ShortcutsGeometryPut
```
Comment (dcoeurjo) [inline ChangeLog.md]: cut/paste issue ?

File: src/DGtal/geometry/surfaces/estimation/LocalEstimatorFromSurfelFunctorAdapter.ih [hunk 2]
```diff
@@ -166,8 +166,13 @@ DGtal::LocalEstimatorFromSurfelFunctorAdapter<TDigitalSurfaceContainer, TMetric,
 eval( const SurfelConstIterator& it ) const
 {
   ASSERT_MSG( isValid(), "Missing init() before evaluation" );
-
+  typedef typename Surface::KSpace KSpace;
   const MetricToPoint metricToPoint = std::bind( *myMetric, myEmbedder( *it ), std::placeholders::_1 );
+  // const RealPoint rbase( myEmbedder( *it ) );
+  // Point     base;
+  // for ( Dimension i; i < KSpace::dimension; ++i )
```
Comment (dcoeurjo) [inline src/DGtal/geometry/surfaces/estimation/LocalEstimatorFromSurfelFunctorAdapter.ih]: cleanup

File: src/DGtal/helpers/Shortcuts.h [hunk 3]
```diff
@@ -1761,6 +1758,108 @@ namespace DGtal
         }
 
 
+      /// Outputs a vector field \a vf anchored at \a
+      /// positions as an OBJ file and a material MTL file. Optionnaly
+      /// you can specify the diffuse colors.
+      ///
+      /// @param[in] positions the bases of the vectors of the vector field.
+      /// @param[in] vf the vector field (an array of real vectors).
```
Comment (dcoeurjo) [inline src/DGtal/helpers/Shortcuts.h]:
```suggestion
      /// @param[in] vf the normal vector field (an array of real vectors).
```

File: src/DGtal/doc/tutorials/moduleShortcuts.dox [hunk 4]
```diff
@@ -430,7 +430,7 @@ Shortcuts::Point is Z3i::KSpace::Point.
   - Shortcuts::getSurfelRange: returns the surfels of a digital surface in the specified traversal order.
   - Shortcuts::getIdxSurfelRange: returns the indexed surfels of an indexed digital surface in the specified traversal order.
   - Shortcuts::getPointelRange: returns the pointels of a digital surface in the default order and optionnaly the map Pointel -> Index giving the indices of each pointel, or simply the pointels around a surfel.
-  - Shortcuts::saveOBJ: saves a digital surface as an OBJ file, with optionally positions, normals and colors information.
+  - Shortcuts::saveOBJ: saves a digital surface as an OBJ file, with optionally positions, normals and colors information. You may also add a vector-field to your OBJ file.
```
Comment (dcoeurjo) [inline src/DGtal/doc/tutorials/moduleShortcuts.dox]: I don't get it.. You can provide normals AND an additional vector field ?
Isn't the second one supposed to be normal as well ?

File: src/DGtal/helpers/Shortcuts.h (hunk 3 above)
Comment (dcoeurjo) [reply src/DGtal/helpers/Shortcuts.h]: as vf is usually used for that

Comment (JacquesOlivierLachaud) [reply src/DGtal/helpers/Shortcuts.h]: It is for displaying *any* vector field. I update the doc.

File: src/DGtal/geometry/surfaces/estimation/LocalEstimatorFromSurfelFunctorAdapter.ih (hunk 2 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/surfaces/estimation/LocalEstimatorFromSurfelFunctorAdapter.ih]: done

File: src/DGtal/doc/tutorials/moduleShortcuts.dox (hunk 4 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/doc/tutorials/moduleShortcuts.dox]: clarified in the doc

File: src/DGtal/helpers/Shortcuts.h [hunk 5]
```diff
@@ -1761,6 +1758,110 @@ namespace DGtal
...
+	      RealPoint    p0 = positions[ i ];
+	      RealPoint    p1 = p0 + vf[ i ];
+	      unsigned int mc = std::max_element( vf.begin(), vf.end() ) - vf.begin();
+	      RealVector   b  =
+		mc == 2 ? RealVector( 1, 0, 0 ) :
+		mc == 1 ? RealVector( 0, 0, 1 ) : RealVector( 0, 1, 0 );
+	      RealVector   e0 = vf[ i ].crossProduct( b ).getNormalized();
+	      RealVector   e1 = vf[ i ].crossProduct( e0 ).getNormalized();
+	      RealPoint    t[4] = {  thickness * e0,  thickness * e1,
+				    -thickness * e0, -thickness * e1 };
```
Comment (dcoeurjo) [inline src/DGtal/helpers/Shortcuts.h]: Well.. I don't understand.. You are not using the `vn` to export the normals ?

Comment (dcoeurjo) [reply src/DGtal/helpers/Shortcuts.h]: This is a bit confusing w.r.t. to the doc and the usage, I think

Comment (dcoeurjo) [reply src/DGtal/helpers/Shortcuts.h]: you're projecting the vertices onto the vf plane ?

Comment (dcoeurjo) [reply src/DGtal/helpers/Shortcuts.h]: ok wait.. I'm confused by the overloading

Comment (dcoeurjo) [reply src/DGtal/helpers/Shortcuts.h]: I would probably have renamed to some specific `saveVectorFieldOBJ`  (not great BTW) and you should probably describe what you mean by "saving a VF as OBJ". Let me check the VCM output

Comment (dcoeurjo) [reply src/DGtal/helpers/Shortcuts.h]: ok.. small tubes..

File: src/DGtal/doc/tutorials/moduleShortcuts.dox [hunk 6]
```diff
@@ -430,7 +430,7 @@ Shortcuts::Point is Z3i::KSpace::Point.
   - Shortcuts::getSurfelRange: returns the surfels of a digital surface in the specified traversal order.
   - Shortcuts::getIdxSurfelRange: returns the indexed surfels of an indexed digital surface in the specified traversal order.
   - Shortcuts::getPointelRange: returns the pointels of a digital surface in the default order and optionnaly the map Pointel -> Index giving the indices of each pointel, or simply the pointels around a surfel.
-  - Shortcuts::saveOBJ: saves a digital surface as an OBJ file, with optionally positions, normals and colors information.
+  - Shortcuts::saveOBJ: several overloaded functions that saves geometric elements as an OBJ file. You may save a digital surface as an OBJ file, with optionally positions, normals and colors information, or may save any vector field as an OBJ file.
```
Comment (dcoeurjo) [inline src/DGtal/doc/tutorials/moduleShortcuts.dox]:
```suggestion
  - Shortcuts::saveOBJ: several overloaded functions that saves geometric elements as an OBJ file. You may save a digital surface as an OBJ file, with optionally positions, normals and colors information, or may save any vector field as an OBJ file (rectangular parallelepipeds aligned with the vectors).
```

File: ChangeLog.md (hunk 1 above)
Comment (JacquesOlivierLachaud) [reply ChangeLog.md]: fixed

File: src/DGtal/helpers/Shortcuts.h (hunk 5 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/helpers/Shortcuts.h]: Done. I have changed the method name.

File: src/DGtal/doc/tutorials/moduleShortcuts.dox (hunk 6 above)
Comment (JacquesOlivierLachaud) [reply src/DGtal/doc/tutorials/moduleShortcuts.dox]: Documentation has been changed to account for saveVectorFieldOBJ

**Comment by dcoeurjo** [comment]
Perfect. thanks

**Pull request merged**